src/effects/effectslot.cpp

```diff
@@ -309,39 +298,29 @@ EffectButtonParameterSlotPointer EffectSlot::getEffectButtonParameterSlot(unsign
     return m_buttonParameters[slotNumber];
 }
 
-void EffectSlot::loadEffectToSlot(EffectsManager* pEffectsManager,
-           EffectManifestPointer pManifest,
-           EffectInstantiatorPointer pInstantiator,
-           const QSet<ChannelHandleAndGroup>& activeChannels) {
+bool EffectSlot::loadEffect(EffectManifestPointer pManifest, EffectInstantiatorPointer pInstantiator,
+        const QSet<ChannelHandleAndGroup>& activeChannels) {
     // Remove an effect if already loaded in slot
     if (m_pEngineEffect != nullptr) {
-        clear();
-        for (int i = 0; i < m_parameters.size(); ++i) {
-            EffectParameter* pParameter = m_parameters.at(i);
-            m_parameters[i] = NULL;
-            delete pParameter;
-        }
-        removeFromEngine();
+        unloadEffect();
     }
 
     m_pManifest = pManifest;
 
-    // Removing the loaded effect
-    if (m_pManifest == nullptr) {
-        return;
+    VERIFY_OR_DEBUG_ASSERT(m_pManifest != nullptr) {
```

Comment:
Should an effect be able to be unloaded by calling loadEffect with a null EffectManifestPointer and/or EffectInstantiatorPointer?